src/os/bluestore/BitAllocator.h

```diff
@@ -0,0 +1,269 @@
+/*
+ * Bitmap based in memory allocator.
+ *
+ * Author: Ramesh Chander , Ramesh.Chander@sandisk.com
+ */
+
+#ifndef __BIT_ALLOCATOR_H
+#define __BIT_ALLOCATOR_H
+
+#include <stdint.h>
+#include <pthread.h>
+
+class BmapEntity {
+public:
+	static int64_t size();
+	virtual int64_t get_index() { return -1; }
+	virtual bool is_allocated(int64_t start, int64_t num) {
+		return true;
+	}
+
+	virtual ~BmapEntity() { }
+};
+
+
+class BmapEntityList {
+	BmapEntity **items;
+	int64_t num_items;
```

Comment:
these few lines are doing exactly what we need. I can tweak them or extend them the way I want. Given I am new to c++, I can start using STL as I feel more comfortable with it?